system/include/emscripten/wasmfs.h

```diff
@@ -22,18 +22,23 @@ backend_t wasmfs_get_backend_by_path(char* path);
 // Obtains the backend_t of a specified fd.
 backend_t wasmfs_get_backend_by_fd(int fd);
 
-// Creates a JSFile Backend in the new file system.
-backend_t wasmfs_create_js_file_backend();
-
-// Creates a Proxied Backend in the new file system.
-backend_t wasmfs_create_proxied_backend(backend_t backend);
-
 // Creates a new file in the new file system under a specific backend.
 uint32_t wasmfs_create_file(char* pathname, mode_t mode, backend_t backend);
 
 // Creates a new directory in the new file system under a specific backend.
 long wasmfs_create_directory(char* path, long mode, backend_t backend);
 
+// Backend creation
+
+// Creates a JSFile Backend in the new file system.
+backend_t wasmfs_create_js_file_backend();
+
+// A function that receives a void* and returns a backend.
+typedef backend_t (*make_backend_t)(void*);
```

Comment:
Maybe `backend_constructor_t`? There are probably other options as well, but it would be good to have a noun phrase rather than a verb phrase for the type name.